camayoc/tests/qpc/ui/test_credentials.py

```diff
@@ -0,0 +1,79 @@
+# coding=utf-8
+"""Tests for handling sources in the UI.
+
+:caseautomation: automated
+:casecomponent: ui
+:caseimportance: high
+:caselevel: integration
+:requirement: Sonar
+:testtype: functional
+:upstream: yes
+"""
+from uuid import uuid4
+
+from selenium.common.exceptions import NoSuchElementException
+
+from smartloc import Locator
+
+from widgetastic.exceptions import WidgetOperationFailed as wof
+from widgetastic.widget import Checkbox
+
+from widgetastic_patternfly import Button, Dropdown
+
+from .utils import clear_toasts, fill
+from .views import CredentialModalView, DashboardView, DeleteModalView
+
+
+def create_credential(view):
+    """Create a credential through the UI."""
+    dash = DashboardView(view)
+    dash.nav.select('Credentials')
+    # TODO: Parametrize for different types of credentials.
+    try:
+        add_credential_dropdown = Dropdown(view, 'Add Credential')
+        add_credential_dropdown.item_select('Network Credential')
+    except NoSuchElementException:
+        add_credential_dropdown = Dropdown(view, 'Add')
+        add_credential_dropdown.item_select('Network Credential')
+    modal = CredentialModalView(view, locator=Locator(css='.modal-content'))
+
+    # workaround, should be `assert modal.save_button.disabled`
+    # https://github.com/RedHatQE/widgetastic.patternfly/pull/66
+    assert modal.save_button.browser.get_attribute(
+            'disabled', modal.save_button)
+    fill(modal, '[placeholder="Enter a name for the credential"]', uuid4())
+    fill(modal, '[placeholder="Enter Username"]', uuid4())
+    fill(modal, '[placeholder="Enter Password"]', uuid4())
+    assert not modal.save_button.browser.get_attribute(
+            'disabled', modal.save_button)
+    modal.save_button.click()
+
+    # clear any artifacts from confirmation dialog, use checkbox as a canary
+    view.wait_for_element(
+            locator=Locator(css='[type="checkbox"]'), delay=0.3)
+    clear_toasts(view=view)
```

Comment:
How are you asserting that the credential was actually created?

Reply:
The checkbox is only present when the credential is created -- that being said, I'll see if I can track the name of the credential and match it that way.